src/main/java/org/wso2/carbon/module/swiftiso20022/utils/JsonToMt103Utils.java

```diff
@@ -0,0 +1,196 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.module.swiftiso20022.utils;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONObject;
+import org.wso2.carbon.module.swiftiso20022.constants.ConnectorConstants;
+import org.wso2.carbon.module.swiftiso20022.constants.MT103Constants;
+import org.wso2.carbon.module.swiftiso20022.validation.JsonToMT103PayloadValidator;
+import org.wso2.carbon.module.swiftiso20022.validation.common.ValidationResult;
+
+
+/**
+ * Class with Json to Mt103 utility methods.
+ */
+public class JsonToMt103Utils {
+
+    private static final Log logger = LogFactory.getLog(JsonToMt103Utils.class);
+
+    /**
+     * Method to validate block 01 of the JSON payload.
+     *
+     * @param block01Json JSON object of block 01. Can be null
+     * @return A {@link ValidationResult} object. Empty if all validations pass, otherwise containing the
+     * first encountered error message.
+     */
+    public static ValidationResult validateBlock01(JSONObject block01Json) {
+
+        logger.debug("Validating block 01");
+
+        // block 01 is mandatory block
+        if (block01Json == null) {
+            return new ValidationResult(ConnectorConstants.ERROR_CODE_MISSING_BLOCK,
+                    String.format(ConnectorConstants.ERROR_MANDATORY_BLOCK_MISSING, MT103Constants.BLOCK01));
+        }
+
+        // if application identifier is present can only contain value "F"
+        if (block01Json.has(MT103Constants.BLOCK01_APPLICATION_ID_KEY)) {
+            if (!MT103Constants.MT103_APPLICATION_ID.equals(
+                    block01Json.getString(MT103Constants.BLOCK01_APPLICATION_ID_KEY))) {
+                return new ValidationResult(ConnectorConstants.ERROR_CODE_INVALID_PARAM,
+                        String.format(ConnectorConstants.ERROR_PARAMETER_INVALID,
+                                ConnectorConstants.BLOCK01_APPLICATION_ID));
+            }
+        }
+
+        // if service identifier is present can only contain value "01"
+        if (block01Json.has(MT103Constants.BLOCK01_SERVICE_ID_KEY)) {
+            if (!MT103Constants.MT103_SERVICE_ID.equals(
+                    block01Json.getString(MT103Constants.BLOCK01_SERVICE_ID_KEY))) {
+                return new ValidationResult(ConnectorConstants.ERROR_CODE_INVALID_PARAM,
+                        String.format(ConnectorConstants.ERROR_PARAMETER_INVALID,
+                                ConnectorConstants.BLOCK01_SERVICE_ID));
+            }
+        }
```

Comment:
Can't we write this logic as a rule and add it to the engine?

Reply:
Add new FixedValueParamValidationRule.java in the commit 4e687a247135e6bc9d43274b4f9f97756cf64480